packages/grid/x-data-grid/src/components/cell/GridEditDateCell.tsx

```diff
@@ -152,6 +152,10 @@ function GridEditDateCell(props: GridEditDateCellProps) {
       value={valueState.formatted}
       onChange={handleChange}
       {...other}
+      sx={{
```

Comment:
`sx` can be a callback
Maybe there is an utility in the core to merge two sx

Reply:
Haven't found anything like that in the core

Reply:
Maybe use `styled(InputBase)` then

Reply:
> Haven't found anything like that in the core

cc @siriwatknp what I hear here is: "please add documentation for it" 😁. Do we have an issue open about it? If not, it could be a great one to open.

The current best practice for the case of Andrew is:

```jsx
  return (
    <InputBase
      inputRef={inputRef}
      fullWidth
      className={classes.root}
      value={valueState.formatted}
      onChange={handleChange}
      {...other}
      sx={[
        {
          fontSize: 'inherit',
        },
        ...(Array.isArray(sx) ? sx : [sx]),
      ]}
    />
  );
}
```

Actually, I think that the DX is also not that great, it's a lot of boilerplate, we could improve the API by flattening automatically like emotion is doing with the styled() API.

Reply:
> I think that we should explore the merits of this capability

+1, otherwise we need to add this in all guides too, as it looks incomplete if we don't have this. cc @siriwatknp what do you think?